optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/value/EntityIndependentValueSelector.java

```diff
@@ -24,4 +26,14 @@
 public interface EntityIndependentValueSelector<Solution_> extends ValueSelector<Solution_>,
         IterableSelector<Solution_, Object> {
 
+    /**
+     * If {@link #isNeverEnding()} is true, then {@link #iterator()} will never end.
+     * This returns an ending {@link Iterator}, that tries to match {@link #iterator()} as much as possible,
+     * but return each distinct element only once
+     * and therefore it might not respect the configuration of this {@link ValueSelector} entirely.
+     *
+     * @return never null
+     * @see #iterator()
+     */
+    Iterator<Object> endingIterator();
```

Comment:
This allows me to easily build an ending iterator for the `ReassignValueToListVariableSelector`.

Reply:
Actually not needed. The `ReassignValueToListVariableSelector` requires that its `childValueSelector` is not never-ending, so it can use the good old `iterator()`.